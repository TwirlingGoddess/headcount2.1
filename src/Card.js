import React from 'react';
import PropTypes from 'prop-types'

const Card = ({index, location, stats, selectDistrict}) => {

	const displayData =
		Object.keys(stats).map((date, index) => {
		if(stats[date] > .5){
			return(
				<li key={index} className="purple">
					{date}: {stats[date]}
				</li>
			)
		} else {
			return(
				<li key={index} className="teal">
					{date}: {stats[date]}
				</li>
			)
		}
	})


	return(
		<div className="Card" onClick={selectDistrict}>
			<h2>{location}</h2>
			<ul>{displayData}</ul>
		</div>
	)
}

Card.propTypes = {
  selectDistrict: PropTypes.func
}

Card.propTypes = {
  location: PropTypes.string
}

Card.propTypes = {
  stats: PropTypes.object
}

Card.propTypes = {
  index: PropTypes.number
}

export default Card