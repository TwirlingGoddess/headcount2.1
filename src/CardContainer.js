import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types'

const CardContainer = ({districts, selectDistrict}) => {
	const deckOfCards = districts.map(district => {
		<Card 	{...district}
				key={district.id}
				selectDistrict={selectDistrict}
		/>
	})

	return(
		<div>
			{deckOfCards}
		</div>
	)
}

export default CardContainer