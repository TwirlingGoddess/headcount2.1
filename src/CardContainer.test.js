import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
	let wrapper
	let mockState
	let expectedId
	let selectMock
	let clickedCard

	beforeEach(() => {
		expectedId = 7
	  selectMock = jest.fn()
		mockState = {location:'Colorado',
										stats:{ 2004: 0.75, 2005: 0.9 }},
									{location:'Juarez',
										stats:{ 2001: 0.15, 2015: 0.29 }}
		wrapper = shallow(<CardContainer location="Colorado" 
					    											districts={mockState}
																		selectDistrict={selectMock}
	    								/>)
		clickedCard = wrapper.find('.Card')
	})

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should render all the ideas in state', () => {
    expect(wrapper.find(Card).length).toEqual(2)
  })	

})