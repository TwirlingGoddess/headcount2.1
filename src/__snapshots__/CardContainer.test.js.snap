// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districts={
        Object {
            "location": "Colorado",
            "stats": Object {
              "2004": 0.75,
              "2005": 0.9,
            },
          }
    }
    location="Colorado"
    selectDistrict={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          location={undefined}
          selectDistrict={[Function]}
          stats={undefined}
/>,
        <Card
          location={undefined}
          selectDistrict={[Function]}
          stats={undefined}
/>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "location": undefined,
          "selectDistrict": [Function],
          "stats": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "location": undefined,
          "selectDistrict": [Function],
          "stats": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            location={undefined}
            selectDistrict={[Function]}
            stats={undefined}
/>,
          <Card
            location={undefined}
            selectDistrict={[Function]}
            stats={undefined}
/>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "location": undefined,
            "selectDistrict": [Function],
            "stats": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "location": undefined,
            "selectDistrict": [Function],
            "stats": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
